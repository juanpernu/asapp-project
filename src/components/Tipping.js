/** React dependencies */
import React from "react";

/** Component dependencies */
import { filterTippingUser } from '../utils';

/**
 * Stateless Component that draw a simple
 * string to know when the other user is tipping.
 * @param {Number} userId
 * @param {Array} users
 */
const Tipping = ({ userId, users }) => {
  return <span className="message-tipping">{filterTippingUser(users, userId)} is tipping...</span>
};

export default Tipping;
