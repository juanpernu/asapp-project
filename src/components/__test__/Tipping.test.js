import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Tipping from '../Tipping';
import MockedUsers from '../../mocks/users'

const mockedData = {
  userId: 0,
  users: MockedUsers
}

describe('Componentes - Tipping', () => {
  it('Should render correctly', () => {
    const { getByText } = render(<Tipping userId={mockedData.userId} users={mockedData.users} />);
    expect(getByText('Laura is tipping...')).toBeInTheDocument();
    expect(Tipping).toMatchSnapshot();
  });
});
