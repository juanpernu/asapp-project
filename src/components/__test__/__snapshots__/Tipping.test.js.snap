// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componentes - Tipping Should render correctly 1`] = `[Function]`;
